### PR TITLE
Hwp 473 moderator view

### PR DESCRIPTION
**Pull request by Bottle7**
# Description

This PR includes the following proposed change(s):

- Determines whether a user is a moderator of the current community
- If the user is a moderator then a 'Flagged Posts' button appears on the right side of the page
- Clicking the 'Flagged Posts' button toggles between seeing all community posts and only flagged posts
- If a post is flagged, a flag icon will be present in the post header next to the title with the tooltip 'Flagged Post'

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Run the app, create a new community, make a few posts, flag some of them, click on 'Flagged Posts' button, see the view switch from all posts to only flagged posts

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [x] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

**Comment by Bottle7** [comment]
The reason that commits aren't squashed in this PR is because when I try to squash I'm getting an unexpected behaviour where I'm seeing commits from outside of the branch during the interactive rebase.

File: app/client/src/views/CommunityPage.jsx [hunk 1]
```diff
@@ -63,14 +65,11 @@ import {
 
 const CommunityPage = (props) => {
   const { communities } = props;
+  console.log(communities);
```
Comment (BradyMitch) [inline app/client/src/views/CommunityPage.jsx:68]:
```suggestion
```

File: app/client/src/views/CommunityPage.jsx [hunk 2]
```diff
@@ -328,6 +356,19 @@ const CommunityPage = (props) => {
...
+              {isModerator === true && (
+                <Box>
+                  <Button
+                    variant="contained"
+                    sx={{
+                      backgroundColor: "error.main",
+                    }}
+                    onClick={handleShowFlaggedPosts}
+                  >
+                    Flagged Posts
```
Comment (BradyMitch) [inline app/client/src/views/CommunityPage.jsx:368]:
```suggestion
                    {showFlaggedPosts ? "Show All Posts" : "Show Flagged Posts"}
```

**Review by BradyMitch — COMMENTED** [review]
Nice work! Once we get isModerator passed down to Post and Comment, I think we should only display the Flag icon if isModerator=true

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**